go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -24,6 +27,15 @@ def parse
 
       private
 
+      # set GOTOOLCHAIN=local if go version >= 1.21
+      def set_gotoolchain_env
+        go_directive = go_mod.content.match(/^go\s(\d+\.\d+)/)&.captures&.first
+        return unless go_directive
```

Comment:
Is it required for `go.mod` files to include a `go` directive? If not, should we set `GOTOOLCHAIN=local` in that case or is it better to keep `GOTOOLCHAIN=go1.20.8`?

Reply:
Surprisingly it is not required, and we do see a small handful of folks who still don't have one set. It's better to leave it for the older 1.20 since 1.21 will try to add a `toolchain` directive.